ext/ddtrace_profiling_native_extension/stack_recorder.c

```diff
@@ -428,7 +430,7 @@ void record_sample(VALUE recorder_instance, ddog_prof_Slice_Location locations,
 
   sampler_unlock_active_profile(active_slot);
 
-  if (result.tag == DDOG_PROF_PROFILE_ADD_RESULT_ERR) {
+  if (result.tag == DDOG_PROF_PROFILE_RESULT_ERR) {
```

Comment:
This was not mentioned in the breaking changes on the PR description. Is this expected?

Reply:
Yes! This was slightly folded under

> A bunch of APIs started returning ddog_prof_Profile_Result.

The API changed to no longer return an `AddResult` and use a regular `Result` (diff of `profiling.h` header):

```diff
-struct ddog_prof_Profile_AddResult ddog_prof_Profile_add(struct ddog_prof_Profile *profile,
+struct ddog_prof_Profile_Result ddog_prof_Profile_add(struct ddog_prof_Profile *profile,
                                                          struct ddog_prof_Sample sample);
```

Error handling on result structure is similar, but the constant to indicate the error is different.